pulp_container/app/registry_api.py

```diff
@@ -684,11 +747,15 @@ def put(self, request, path, pk=None):
         """
         Create a blob from uploaded chunks.
 
-        Note: We do not support monolithic upload.
+        This request makes the upload complete. It can whether carry a zero-length
```

Comment:
here, this can be tested with podman( which is sending empty body in the PUT request) and helm client that sends last ( in its case the only chunk) in the PUT request.

Reply:
And, the last chunk can also be the "main" chunk in the monolithic upload, right?

>  To carry out a “monolithic” upload, one can simply put the entire content blob to the provided URL: PUT /v2/<name>/blobs/uploads/<uuid>?digest=\<digest\>

https://docs.docker.com/registry/spec/api/#monolithic-upload

Reply:
In the monolithic upload it is always the last and main chunk, that's why it is called monolithic.

It never issues a PATCH request where chunks are being send ( as in chunked upload).
To make it even more confusing, podman claims to use chunked upload, but in reality it sens only one chunk in the PATCH request and no body in the PUT. IT also does not send the content-range header in the PATCH request that would indicate a legal chunked upload ;)